src/WebJobs.Script.WebHost/WebHostServiceCollectionExtensions.cs

```diff
@@ -128,13 +128,27 @@ public static void AddWebJobsScriptHost(this IServiceCollection services, IConfi
             services.AddSingleton<IFunctionMetadataManager, FunctionMetadataManager>();
             services.AddSingleton<IFunctionMetadataProvider, HostFunctionMetadataProvider>();
             services.AddSingleton<IWebFunctionsManager, WebFunctionsManager>();
-            services.AddSingleton<IInstanceManager, InstanceManager>();
             services.AddHttpClient();
             services.AddSingleton<StartupContextProvider>();
             services.AddSingleton<IFileSystem>(_ => FileUtility.Instance);
             services.AddTransient<VirtualFileSystem>();
             services.AddTransient<VirtualFileSystemMiddleware>();
 
+            if (SystemEnvironment.Instance.IsLinuxConsumptionOnAtlas())
+            {
+                System.Console.WriteLine("[TEST] Atlas Linux Consumption");
+                services.AddSingleton<LinuxInstanceManager, InstanceManager>();
```

Comment:
We should retain registering this as `IInstanceManager`. Same with the legion registration below.
```suggestion
                services.AddSingleton<IInstanceManager, InstanceManager>();
```